RecoLocalCalo/HcalRecProducers/src/HFPreReconstructor.cc

```diff
@@ -138,16 +142,14 @@ void HFPreReconstructor::fillInfos(const edm::Event& e, const edm::EventSetup& e
   // Clear the collection we want to fill in this method
   qie10Infos_.clear();
 
-  // Get the Hcal topology if needed
-  ESHandle<HcalTopology> htopo;
-  if (tsFromDB_) {
-    eventSetup.get<HcalRecNumberingRecord>().get(htopo);
-    paramTS_->setTopo(htopo.product());
-  }
-
   // Get the calibrations
-  ESHandle<HcalDbService> conditions;
-  eventSetup.get<HcalDbRecord>().get(conditions);
+  ESHandle<HcalTopology> htopoHandle;
+  eventSetup.get<HcalRecNumberingRecord>().get(htopoHandle);
+  const HcalTopology& htopo(*htopoHandle);
+
+  ESHandle<HcalChannelPropertiesVec> propHandle;
+  eventSetup.get<HcalChannelPropertiesRecord>().get(propHandle);
+  const HcalChannelPropertiesVec& prop(*propHandle);
```

Comment:
Could these be changed to use the `ESGetToken`?
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#In_ED_module
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#Getting_data_from_EventSetup_wit

Reply:
Fixes of this kind are outside of the purpose and scope of this PR. They need to be done for many modules, and it would be best if these changes are performed separately.

Reply:
Separate PR is perfectly fine, thanks.